bigquery/storage/managedwriter/client.go

```diff
@@ -250,13 +249,14 @@ func (c *Client) createPool(ctx context.Context, settings *streamSettings, strea
 	}
 
 	pool := &connectionPool{
-		ctx:                  cCtx,
-		allowMultipleWriters: allowMultipleWriters,
-		cancel:               cancel,
-		open:                 createOpenF(ctx, streamFunc),
-		callOptions:          arOpts,
-		baseFlowController:   newFlowController(fcRequests, fcBytes),
+		id:                 newUUID(poolIDPrefix),
+		ctx:                cCtx,
+		cancel:             cancel,
+		open:               createOpenF(ctx, streamFunc),
+		callOptions:        arOpts,
+		baseFlowController: newFlowController(fcRequests, fcBytes),
 	}
+	router := newSharedRouter(c.cfg.useMultiplex, c.cfg.maxMultiplexPoolSize)
 	if err := pool.activateRouter(router); err != nil {
```

Comment:
why activateRouter is not passed in to constructor?

Reply:
Formerly, we did it this way.  We now use a method for attaching to allow the two components a chance too validate and build references.